addons/preview-project-description/userscript.js

```diff
@@ -0,0 +1,92 @@
+  enableSwitcher.classList.add("button", "action-button", "sa-preview-desc");
+  enableSwitcher.addEventListener("click", togglePreview);
+  enableSwitcher.appendChild(enableSwitcherText);
+
+  const instructionPreview = document.createElement("div");
+  const instructionEditor = document.querySelector('textarea[name="description"]');
+  const instructionContainer = instructionEditor.closest(".description-block");
+  const instructionForm = instructionEditor.closest(".project-description-form");
+  instructionPreview.classList.add("project-description");
+  instructionPreview.innerText = instructionEditor.value;
+
+  const notesCreditPreview = document.createElement("div");
+  const notesCreditEditor = document.querySelector('textarea[name="instructions"]');
+  const notesCreditContainer = notesCreditEditor.closest(".description-block");
+  const notesCreditForm = notesCreditEditor.closest(".project-description-form");
+  notesCreditPreview.classList.add("project-description");
+  notesCreditPreview.innerText = notesCreditEditor.value;
+
+  notesCreditContainer.appendChild(notesCreditPreview);
+  instructionContainer.appendChild(instructionPreview);
+  actionsContainer.appendChild(enableSwitcher);
+
+  var currentlyEnabled = addon.settings.get("enable-default");
+  togglePreview(null, currentlyEnabled);
+
+  /**
+   * Will attempt to toggle between the off and on states, however an override can be passed which will completely override the current value
+   * @param {any} _ This exists because this is used as an event body(line 13)
+   * @param {boolean} override Force to a specific value.
+   * @returns {void}
+   */
+  function togglePreview(_ = null, override = !currentlyEnabled) {
+    enableSwitcherText.innerText = override ? msg("Disable") : msg("Enable");
+    currentlyEnabled = override;
+
+    if (override) {
+      parseEditorInput(instructionPreview, instructionEditor);
+      parseEditorInput(notesCreditPreview, notesCreditEditor);
+    }
+
+    setDisplayable(instructionForm, override);
+    setDisplayable(instructionPreview, !override);
+
+    setDisplayable(notesCreditForm, override);
+    setDisplayable(notesCreditPreview, !override);
+  }
+
+  /**
+   *
+   * @param {HTMLDivElement} preview Where to place the rendered output.
+   * @param {Element} editor Where to grab the input to render, should be an editable with a value attribute.
+   */
+  async function parseEditorInput(preview, editor) {
+    let input = editor.value.replace(/\</g, "&lt;").replace(/\>/g, "&gt;").replace(/\&/g, "&amp;");
+    let rendered = input.replace(matchUsername, (matched) => {
+      return `<a href="/users/${matched.slice(1)}">${matched}</a>`;
+    });
+    if (moreLinksEnabled)
+      rendered.replace(matchHyperLinks, (matched) => {
+        if (matched.startsWith("https://") || matched.startsWith("http://")) matched = "//" + matched;
+        return `<a href="${matched}">${matched}</a>`;
+      });
+    preview.innerHTML = rendered;
```

Comment:
This is probably vulnerable to XSS

Reply:
Maybe, but at the top of the function I'm matching for `<`, `>`, and `&`. I could probably also add `"` can't escape out of `href`, but otherwise I don't see how else it could be XSS unless you're talking about the site it brings you to.